core/src/Kokkos_Graph.hpp

```diff
@@ -161,6 +161,11 @@ Graph<ExecutionSpace> create_graph(Closure&& arg_closure) {
 #include <impl/Kokkos_GraphNodeImpl.hpp>
 #include <impl/Kokkos_Default_Graph_Impl.hpp>
 #include <Cuda/Kokkos_Cuda_Graph_Impl.hpp>
+#if defined(KOKKOS_ENABLE_HIP)
+#if !((HIP_VERSION_MAJOR == 5) && (HIP_VERSION_MINOR == 2))
+#include <HIP/Kokkos_HIP_Graph_Impl.hpp>
```

Comment:
Hi @Rombur ! I can't wait to see the HIP implementation of `Kokkos::Graph`!

I'm just a bit worried about the code duplication here. As you mentioned in the PR description, the HIP API is very similar to the one of CUDA.

So I wonder if having some sort of macro for the common part would help. Some classes are cleary near copy-pastes.

For some functions like `create_aggregate_ptr` it's even more obvious...

Reply:
It's true that the code is very similar but it's always the case when we first port CUDA code to HIP. Over time the backends are optimized differently and the code diverges.